pytest_html/plugin.py

```diff
@@ -127,18 +131,36 @@ def __lt__(self, other):
                      'XPassed', 'Skipped', 'Passed')
             return order.index(self.outcome) < order.index(other.outcome)
 
-        def append_extra_html(self, extra, additional_html, links_html):
+        def append_extra_html(self, extra):
             href = None
             if extra.get('format') == extras.FORMAT_IMAGE:
-                href = '#'
-                image = 'data:image/png;base64,{0}'.format(
-                        extra.get('content'))
-                additional_html.append(html.div(
-                    html.a(html.img(src=image), href="#"),
+                image_src = None
+                if self.contained_html:
+                    image_src = 'data:image/png;base64,{0}'.format(
+                            extra.get('content'))
+                    href = '#'
+                else:
+                    hash_key = self.test_id + self.outcome
+                    hash_generator = hashlib.md5()
+                    hash_generator.update(hash_key)
+                    image_file_name = '{0}.jpg'.format(hash_generator.digest)
+
+                    super_dir_name = os.path.dirname(self.logfile)
+                    dir_name = os.path.join(super_dir_name, 'assets')
+                    image_path = os.path.join(dir_name, image_file_name)
+                    href = os.path.join('assets', image_file_name)
+                    image_src = href
+
+                    with open(image_path, 'w') as fh:
```

Comment:
We should probably create the assests directory in case it doesn't already exist. We do this in `_save_report` already, but that's too late for writing the images to it.